fahrschein-http-jdk11/src/main/java/org/zalando/fahrschein/http/jdk11/JavaNetBufferingRequest.java

```diff
@@ -68,7 +68,7 @@ public void add(String headerName, String value) {
 
             @Override
             public void put(String headerName, String value) {
-                request.header(headerName, value);
+                request.setHeader(headerName, value);
```

Comment:
You are changing the behavior here. Is this actually intended? `setHeader` overwrites, whereas `header` adds?

Reply:
yes, that is fixing a bug because we have `add(headerName, value)` and `put(headerName, value)` - even though there's no javadoc explaining the difference, `put` should overwrite, while `add` should append.